src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SyntaxNodeExtensions.cs

```diff
@@ -92,5 +92,129 @@ public static Range GetRange(this SyntaxNode node, RazorSourceDocument source)
 
             return range;
         }
+
+        public static Range? GetRangeWithoutWhitespace(this SyntaxNode node, RazorSourceDocument source)
+        {
+            if (node is null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            var tokens = node.GetTokens();
+
+            SyntaxToken firstToken = null;
+            for (var i = 0; i < tokens.Count; i++)
+            {
+                var token = tokens[i];
+                if (!IsWhitespace(token))
+                {
+                    firstToken = token;
+                    break;
+                }
+            }
+
+            SyntaxToken lastToken = null;
+            for (var i = tokens.Count - 1; i >= 0; i--)
+            {
+                var token = tokens[i];
+                if (!IsWhitespace(token))
+                {
+                    lastToken = token;
+                    break;
+                }
+            }
+
+            if (firstToken is null && lastToken is null)
+            {
+                return null;
+            }
+
+            var startPositionSpan = GetLinePositionSpan(firstToken, source, node.SpanStart);
+            var endPositionSpan = GetLinePositionSpan(lastToken, source, node.Span.Start);
```

Comment:
Whats the difference between `node.SpanStart` and `node.Span.Start`?

Reply:
None, it was a silly typo that just happened to work.